mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Vector is little-endian encoded as follows:
+//   0 -  8  number of elements     (int64)
+//   8 - ..  vector's data elements (float64)
+func (v Vector) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeInt64+v.n*sizeFloat64))
+	err := binary.Write(buf, defaultEndian, int64(v.n))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, val := range v.mat.Data {
```

Comment:
I guess my main worry is that breaking the invariant `v === unmarshal(marshal(v))` where `===` is the identity (not just mere equality) could break people's code in interesting ways and bite us in the rear area down the road.

besides, the current implementation *is* cross-platform and cross-language friendly.
a `python` module can easily read the resulting binary file and discard, if deemed unrelevant, the extra `Inc` int64 (that I would, of course, need to add to the binary layout)